app/main.py

```diff
@@ -1,2 +1,68 @@
+from typing import Any, Optional
+from dataclasses import dataclass
+
+
+@dataclass
+class Node:
+    key: Any
+    value: Any
+    _next: Optional["Node"] = None
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity: int = 8
+        self.hash_table: list[Optional[Node]] = [None] * self.capacity
+        self.length: int = 0
+
+    def __setitem__(self, key: Any, value: Any) -> None:
```

Comment:
also key is not Any type, it's a Hashable type from typing lib, not Any can be hashable